src/components/cards/EthCard.tsx

```diff
@@ -57,27 +57,24 @@ export const EthCard = () => {
     emitChartsRequest([ETH_ADDRESS], chartTypes.day, nativeCurrency);
   }, [nativeCurrency]);
 
-  const handlePressBuy = () => {
-    if (isDamaged) {
-      showWalletErrorAlert();
-      return;
-    }
+  const handlePressBuy = useCallback(
+    e => {
+      e.stopPropagation();
```

Comment:
whats the reasoning here?

Reply:
Otherwise it bubbles up to the next onPress handler that opens the token sheet.